### PR TITLE
admin can edit delete questions

### PR DESCRIPTION
**Pull request by thanhnga67**

### CODE REVIEW

File: app/controllers/admin/questions_controller.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class Admin::QuestionsController < Admin::BaseController
-  before_action :set_category, only: [:new, :create]
+  load_and_authorize_resource
+  before_action :set_category
```
Comment (VoTaiTri) [inline app/controllers/admin/questions_controller.rb]: filter

File: app/views/admin/questions/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<% provide :title, t("edit") %>
+<h1><%= t "edit" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= render "form", button: t("edit")%>
```
Comment (leminhtuan2015) [inline app/views/admin/questions/edit.html.erb]: This is not a comment, I just share something

- You dont need to pass the "button" to the _form.html to show "create" or "edit" for submit button.

- you can check current controller in _form.html
http://stackoverflow.com/questions/3757491/can-i-get-the-name-of-the-current-controller-in-the-view

Thankis.

Comment (thanhnga67) [inline app/views/admin/questions/edit.html.erb]: :) thank you so much

File: app/controllers/admin/questions_controller.rb (hunk 1 above)
Comment (thanhnga67) [inline app/controllers/admin/questions_controller.rb]: dạ cái set_category action nào e cũng dùng nên e để vậy ạ

File: app/controllers/admin/questions_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class Admin::QuestionsController < Admin::BaseController
-  before_action :set_category, only: [:new, :create]
-
+  load_and_authorize_resource
+  before_action :set_category
```
Comment (yuyuvn) [inline app/controllers/admin/questions_controller.rb]: try this:

load_and_authorize_resource :category
load_and_authorize_resource through: :category, shallow: true

File: app/controllers/admin/questions_controller.rb (hunk 1 above)
Comment (VoTaiTri) [inline app/controllers/admin/questions_controller.rb]: def new với create cũng cần cái này ah?

File: app/views/admin/questions/_form.html.erb [hunk 4]
```diff
@@ -17,5 +17,5 @@
     </p>
   </div>
 
-  <%= f.submit t("create"), class: "btn btn-default btn-lg" %>
+  <%= f.submit controller.action_name, class: "btn btn-default btn-lg" %>
```
Comment (yuyuvn) [inline app/views/admin/questions/_form.html.erb]: you still need i18n
how about "t(controller.action_name)" ?

File: app/controllers/admin/questions_controller.rb (hunk 1 above)
Comment (thanhnga67) [inline app/controllers/admin/questions_controller.rb]: vâng ạ. khi new question thì có lưu cả id của category ạ

File: app/controllers/admin/questions_controller.rb (hunk 3 above)
Comment (thanhnga67) [inline app/controllers/admin/questions_controller.rb]: :+1: Thank yo so much :*

**Pull request merged**